### PR TITLE
Review Code

### PR DESCRIPTION
**Pull request by osama-maree**

### CODE REVIEW

File: src/calculator.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const calc = (...arg) => {
```
Comment (m5atib) [inline src/calculator.js]: Choosing a good name for a function is important for code readability and understanding its purpose. So write a well-name for this function.

File: src/calculator.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  if (arg.length < 3 || !(arg.length % 2)) {
+    throw new Error("Invalid number of inputs");
+  }
+  // Ignore vale more than 1000
+    for (let i = 0; i < arg.length; i += 2) {
```
Comment (m5atib) [inline src/calculator.js]: Please use javascript Array built-in methods like `filter` , `map` , `forEach` , etc

File: src/calculator.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      if (arg[i] > 1000 && i == 0) {
+        arg.splice(i, 2);
+        i -= 2;
+      } else if (arg[i] > 1000) {
+        arg.splice(i - 1, 2);
+        i -= 2;
+      }
+    }
+  const equ = [];
+  for (let i = 1; i < arg.length; i += 2) {
```
Comment (m5atib) [inline src/calculator.js:23]: Please use javascript Array built-in methods like filter , map , forEach , etc

File: src/calculator.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      }
+      equ.push(arg[i - 1]);
+      equ.push(arg[i]);
+    } else {
+      arg[i + 1] = evaluate(arg[i - 1], arg[i + 1], arg[i]);
+    }
+  }
+  if (equ.length == 0) return arg[arg.length - 1];
+  if (typeof equ[equ.length - 1] !== "number") equ.push(arg[arg.length - 1]);
+  for (let i = 1; i < equ.length; i += 2) {
```
Comment (m5atib) [inline src/calculator.js]: Please use javascript Array built-in methods like filter , map , forEach , etc

**Review by m5atib — CHANGES_REQUESTED** [review]
Please do the needed changes

File: src/calculator.js (hunk 1 above)
Comment (osama-maree) [reply src/calculator.js]: Well, Than you

File: src/calculator.js (hunk 2 above)
Comment (osama-maree) [reply src/calculator.js]: Well, I will try to solve it using them

File: src/calculator.js (hunk 3 above)
Comment (osama-maree) [reply src/calculator.js:23]: Well, I will try to solve it using them

File: src/calculator.js (hunk 4 above)
Comment (osama-maree) [reply src/calculator.js]: Well, I will try to solve it using them

**Pull request merged**